src/styles/main.scss

```
@@ -1,3 +1,58 @@
+$day-size: 100px;
+$calendar-size: 7 * $day-size + 6px;
+$days: "mon" 1, "tue" 2, "wed" 3, "thu" 4, "fri" 5, "sat" 6, "sun" 7;
+
 body {
   margin: 0;
+  font-family: Arial, Helvetica, sans-serif;
+  font-size: 30px;
+}
+
+.calendar {
+  display: flex;
+  flex-wrap: wrap;
+  gap: 1px;
+  width: $calendar-size;
+  padding: 30px 10px;
+
+  @each $first-day, $index in $days {
+    &--start-day--#{$first-day} {
+      .day:first-child {
+        margin-left: ($index - 1) * ($day-size + 1px);
+      }
+    }
+  }
+
+  @for $index from 28 through 31 {
+    $d: $index + 1;
+    &--month-length--#{$index} {
+      .day:nth-child(n + #{$d}) {
+        visibility: hidden;
+      }
+    }
+  }
+}
+
+.day {
+  box-sizing: border-box;
+  height: $day-size;
+  width: $day-size;
+  border: 1px solid black;
+  background-color: #eee;
+  display: flex;
+  align-items: center;
+  justify-content: center;
+  cursor: pointer;
+
+  @for $index from 1 through 31 {
+    &:nth-child(#{$index})::before {
+      content: "#{$index}";
+    }
+  }
+
+  &:hover {
+    background-color: pink;
+    transform: translateY(-20px);
+    transition: 0.5s;
```

```suggestion
    transition: 0.5s;
```
властивість transition потрібно прписувати не на хавер, а на сам елемент. Тоді буде плавана анімація в обидва боки. Зараз, якщо прибрати мишку з елемента, він різко, без анімації, повертається назад